packages/xod-client/src/editor/containers/Tabs.jsx

```diff
@@ -75,8 +73,12 @@ class Tabs extends React.Component {
     setTimeout(() => this.props.actions.closeTab(tabId), 0);
   }
 
-  onSortEnd(changes) {
-    const sortedTabs = swap(changes.oldIndex, changes.newIndex, this.getTabs());
+  onSortEnd({ oldIndex, newIndex }) {
+    const tabs = this.getTabs();
+    const sortedTabs = R.compose(
+      R.insert(newIndex, tabs[oldIndex]),
+      R.remove(oldIndex, 1)
+    )(tabs);
     const newTabs = assocIndexes(sortedTabs);
     const indexedTabs = indexById(newTabs);
```

Comment:
No need for these intermediate constants anymore, it all can be in one `compose`

Reply:
Done!